x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDateMappedFieldType.java

```diff
@@ -86,6 +88,30 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
         return scriptFactory.newFactory(script.getParams(), lookup);
     }
 
+    @Override
+    public Query distanceFeatureQuery(Object origin, String pivot, float boost, QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
+        return DateFieldType.handleNow(context, now -> {
+            long originLong = DateFieldType.parseToLong(
+                origin,
+                true,
+                null,
+                dateTimeFormatter.toDateMathParser(),
+                now,
+                DateFieldMapper.Resolution.MILLISECONDS
+            );
+            TimeValue pivotTime = TimeValue.parseTimeValue(pivot, "distance_feature.pivot");
+            return new LongScriptFieldDistanceFeatureQuery(
+                script,
+                leafFactory(context.lookup())::newInstance,
+                name(),
+                originLong,
+                pivotTime.getMillis(),
+                boost
+            );
+        });
+    }
```

Comment:
I wonder what the plan is for the instanceof checks in DistanceFeatureQueryBuilder#doToQuery . Are we ok with keeping those?

Reply:
oh actually those are gone upstream, great! sorry for the noise then, you already did what I would asked you to do